src/reference/asciidoc/kafka.adoc

```diff
@@ -2438,9 +2439,58 @@ This type inference can be achieved only when the `@KafkaListener` annotation is
 With a class-level `@KafkaListener`, the payload type is used to select which `@KafkaHandler` method to invoke, so it must already have been converted before the method can be chosen.
 ====
 
-NOTE: When you use the `StringJsonMessageConverter`, you should use a `StringDeserializer` in the Kafka consumer configuration and a `StringSerializer` in the Kafka producer configuration when you use Spring Integration or the `KafkaTemplate.send(Message<?> message)` method.
-When you use the `BytesJsonMessageConverter`, you should use a `BytesDeserializer` in the Kafka consumer configuration and `BytesSerializer` in the Kafka producer configuration when you use Spring Integration or the `KafkaTemplate.send(Message<?> message)` method (see <<kafka-template>>).
-Generally, the `BytesJsonMessageConverter` is more efficient because it avoids a `String` to and from `byte[]` conversion.
+[NOTE]
+====
+On the consumer side, you can configure a `JsonMessageConverter`; it can handle `ConsumerRecord` values of type `byte[]`, `Bytes` and `String` so should be used in conjunction with a `ByteArrayDeserializer`, `BytesDeserializer` or `StringDeserializer`.
+(`byte[]` and `Bytes` are more efficient because they avoid an unecessary `byte[]` to `String` conversion).
+You can also configure the specific subclass of `JsonMessageConverter` corresponding to the deserializer, if you so wish.
+
+On the producer side, when you use Spring Integration or the `KafkaTemplate.send(Message<?> message)` method (see <<kafka-template>>), you must configure a message converter that is compatible with the configured Kafka `Serializer`.
+
+* `StringJsonMessageConverter` with `StringSerializer`
+* `BytesJsonMessageConverter` with `BytesSerializer`
+* `ByteArrayJsonMessageConverter` with `ByteArraySerializer`
+
+Again, using `byte[]` or `Bytes` is more efficient because they avoid a `String` to `byte[]` conversion.
+====
+
+[[data-projection]]
+====== Using Spring Data Projection Interfaces
+
+Starting with version 2.1.1, you can convert JSON to a Spring Data Projection interface instead of a concrete type.
+This allows very selective, and low-coupled bindings to data, including the lookup of values from multiple places inside the JSON document.
+For example the following interface can be defined as message payload type:
+
+====
+[source, java]
+----
+interface SomeSample {
+
+  @JsonPath({ "$.username", "$.user.name" })
+  String getUsername();
+
+}
+----
+====
+
+====
+[source, java]
+----
+@KafkaListener(id="projection.listener", topics = "projection")
+public void projection(SomeSample in) {
+    String username = in.getUsername();
+    ...
+}
+----
+====
+
+Accessor methods will be used to lookup the property name as field in the received JSON document by default.
+The @JsonPath expression allows customization of the value lookup, and even to define multiple JSONPath expressions, to lookup values from multiple places until an expression returns an actual value.
```

Comment:
Let's wrap `@JsonPath` to the code? For consistency with `@KafkaListener` around.